deftree/deftree.go

```diff
@@ -136,23 +136,40 @@ func (self *MicroserviceDefinition) GetByName(name string) Describable {
 	return nil
 }
 
+type optionErr struct {
```

Comment:
IIRC, we managed to remove a lot of the optional error usages throughout the repo in previous changes. Personally I'd rather define a error in this package and check the Type in the consuming function over incorporating the optional error again.